server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java

```diff
@@ -240,6 +243,39 @@ protected void doStop() {
             }
         }
 
+        long gracefulShutdownMillis = SETTING_HTTP_GRACEFUL_SHUTDOWN.get(settings).getMillis();
+        if (gracefulShutdownMillis > 0 && httpChannels.size() > 0) {
+
+            logger.info(
+                "There are {} open client connections, try to close gracefully within {}ms.",
+                httpChannels.size(),
+                gracefulShutdownMillis
+            );
+
+            // Close connection after the client is getting a response.
+            handlingSettings.setForceCloseConnection();
```

Comment:
Sorry @tmanninger , getting to it again just now. So I think this is what we want:
 - try to send the "in-flight" response to the clients within `gracefulShutdownMillis` timeframe
 - close the connection (forceably or not)

I believe we could make it part of the `RestChannel` (and leave `AbstractHttpServerTransport` out of it) by introducing `RestChannel::closeGracefuly(long timeout)` method and calling it in `CloseableChannel.closeChannels` (would need some adjustments).